serverless.yml

```diff
@@ -1,93 +1,157 @@
+
+  customDomain:
+    domainName: ${opt:domain}
+    basePath: 'api'
+    stage: ${self:provider.stage}
+    createRoute53Record: true
+
 provider:
   name: aws
   runtime: java8
-  stage: dev
-  region: us-east-1
-
-# you can add statements to the Lambda function's IAM Role here
-#  iamRoleStatements:
-#    - Effect: "Allow"
-#      Action:
-#        - "s3:ListBucket"
-#      Resource: { "Fn::Join" : ["", ["arn:aws:s3:::", { "Ref" : "ServerlessDeploymentBucket" } ] ]  }
-#    - Effect: "Allow"
-#      Action:
-#        - "s3:PutObject"
-#      Resource:
-#        Fn::Join:
-#          - ""
-#          - - "arn:aws:s3:::"
-#            - "Ref" : "ServerlessDeploymentBucket"
-#            - "/*"
-
-# you can define service wide environment variables here
-#  environment:
-#    variable1: value1
-
-# you can add packaging information here
+  stage: ${opt:stage, 'dev'}
+  region: ${opt:region, 'us-east-1'}
+
+  iamRoleStatements:
+    - Effect: "Allow"
+      Action:
+        - dynamodb:Query
+        - dynamodb:Scan
+        - dynamodb:GetItem
+        - dynamodb:PutItem
+        - dynamodb:UpdateItem
+        - dynamodb:DeleteItem
+      Resource:
+        - { "Fn::GetAtt": ["surveyTable", "Arn"] }
+        - { "Fn::GetAtt": ["questionTable", "Arn"] }
+        - { "Fn::GetAtt": ["answerTypeTable", "Arn"] }
+        - { "Fn::GetAtt": ["emailTable", "Arn"] }
+
+    - Effect: "Allow"
+      Action:
+        - "ses:SendEmail"
+        - "ses:SendRawEmail"
+      Resource: "*"
+
+  environment:
+    SURVEY_TABLE_NAME: ${self:custom.surveyTableName}
+    QUESTION_TABLE_NAME: ${self:custom.questionTableName}
+    ANSWER_TYPE_TABLE_NAME: ${self:custom.answerTypeTableName}
+    EMAIL_TABLE_NAME: ${self:custom.emailTableName}
+
+
 package:
-  artifact: target/feedback-survey-SNAPSHOT-1.jar
+  artifact: target/feedback-survey-1.jar
 
 functions:
-  hello:
-    handler: com.buildit.serverless.Handler
+  saveSurvey:
+    handler: com.buildit.survey.SaveSurveyHandler
+    timeout: 30
+    events:
+      - http:
+          path: /saveSurvey
+          method: post
+          cors: true
+
+  getQuestions:
+    handler: com.buildit.question.GetQuestionsHandler
     events:
       - http:
-          path: /hello
+          path: /getQuestions
           method: get
+          cors: true
 
-  saveSurvey:
-    handler: com.buildit.survey.SaveSurveyHandler
+  saveQuestions:
+    handler: com.buildit.question.SaveQuestionsHandler
     events:
       - http:
-          path: /saveSurvey
+          path: /saveQuestions
+          method: post
+          authorizer: aws_iam
+
+  saveAnswerTypes:
+    handler: com.buildit.question.SaveAnswerTypesHandler
+    events:
+      - http:
+          path: /saveAnswerTypes
           method: post
+          authorizer: aws_iam
+
+  saveEmail:
+    handler: com.buildit.email.SaveEmailHandler
+    events:
+      - http:
+          path: /saveEmail
+          method: post
+          authorizer: aws_iam
+
+resources:
```

Comment:
best would be to refactor this table structure as this is not a `RMDB` but `NoSQL`, 

from AWS docs

> As a general rule, you should maintain as few tables as possible in a DynamoDB application. As emphasized earlier, most well designed applications require only one table, unless there is a specific reason for using multiple tables.

[source](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/bp-general-nosql-design.html)